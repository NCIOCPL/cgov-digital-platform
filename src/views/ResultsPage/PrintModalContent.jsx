import React, { useEffect } from 'react';
import {useSelector} from 'react-redux';
import { usePrintApi } from '../../utilities/hooks';
import { history } from '../../services/history.service';

const PrintModalContent = ({ selectedList = [], handleClose = () => {} }) => {
  // in dev use
  const printUrl = useSelector(store => store.globals.printCacheEndpoint);

  const [{ data, isLoading, isError, doPrint }] = usePrintApi(
    { TrialIDs: selectedList },
    printUrl
  );

  // on component mount, check for selected IDs,
  useEffect(() => {
    if (selectedList.length > 0 && selectedList.length <= 100) {
      doPrint();
    }
  }, [selectedList]);

  useEffect(() => {
    if (!isLoading && !isError && data.printID) {
      //success
      closeModal();
      history.push({
        pathname: '/CTS.Print/Display',
        search: '?printid=' + data.printID,
      });
    }
    if (isError) {
      console.log('Error occurred.');
    }
  }, [isLoading, isError, data]);

  const closeModal = () => {
    handleClose();
  };

  const renderNoItemsSelected = () => (
    <>
      <div className="icon-warning" aria-hidden="true">
        !
      </div>
      <p>
        You have not selected any trials. Please select at least one trial to
        print.
      </p>
    </>
  );
  const renderPrintError = () => (
    <>
      <div className="icon-warning" aria-hidden="true">
        !
      </div>
      <p>
        An error occurred while generating your document. Please try again
        later.
      </p>
    </>
  );

  const renderTooManyItemsSelected = () => (
    <>
      <div className="icon-warning" aria-hidden="true">
        !
      </div>
      <p>
        You have selected the maximum number of clinical trials (100) that can
        be printed at one time.
      </p>
      <p>
        Print your current selection and then return to your search results to
        select more trials to print.
      </p>
    </>
  );

  const renderPrintInterstitial = () => (
    <>
      <div className="spinkit spinner">
        <div className="dot1"></div>
        <div className="dot2"></div>
      </div>
      <p>
        You will automatically be directed to your print results in just a
        moment...
      </p>
    </>
  );

  return (
    <>
      {selectedList.length === 0 ? (
        renderNoItemsSelected()
      ) : selectedList.length >= 100 ? (
        renderTooManyItemsSelected()
      ) : (
        <>
          {isLoading && <>{renderPrintInterstitial()}</>}
          {isError && <>{renderPrintError()}</>}
        </>
      )}
    </>
  );
};

export default PrintModalContent;
