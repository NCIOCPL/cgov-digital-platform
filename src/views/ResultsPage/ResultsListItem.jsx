import React from 'react';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Checkbox from '../../components/atomic/Checkbox';
import { isWithinRadius } from '../../utilities/utilities';

const ResultsListItem = ({ id, item, isChecked, onCheckChange }) => {
  const {
    zip,
    zipCoords,
    zipRadius,
    location,
    country,
    states,
    city,
  } = useSelector(store => store.form);

  //compare site values against user criteria
   const isLocationParamMatch = siteObj => {
    if (siteObj.country === country) {
      if (country === 'United States') {
        if (states.length > 0) {
          const statesList = [
            ...new Set(states.map(item => item.abbr)),
          ];
          if (statesList.includes(siteObj.stateOrProvinceAbbreviation)) {
            if (siteObj.city && siteObj.city === city) {
              return true;
            } 
            return true;
          } else {
            return false;
          }
        }
        if (city !== '') {
          if (siteObj.city === city) {
            return true;
          } else {
            return false;
          }
        }
        // only looking for US sites
        return true;
      } else {
        // check for city
        if (city !== '') {
          if (siteObj.city === city) {
            return true;
          } else {
            return false;
          }
        } else {
          // only searching on country but is match
          return true;
        }
      }
    } else {
      return false;
    }
  };

  const countNearbySitesByZip = arr => {
    return arr.reduce(
      (count, siteItem) =>
        count + isWithinRadius(zipCoords, siteItem.coordinates, zipRadius),
      0
    );
  };

  const countNearbySitesByCountryParams = arr => {
    return arr.reduce(
      (count, siteItem) => 
        count + isLocationParamMatch(siteItem),
      0
    );
  };

  const getGenderDisplay = genderVal => {
    const displays = {
      MALE: 'Male',
      FEMALE: 'Female',
      BOTH: 'Male or Female',
    };
    return displays[genderVal];
  };

  const getAgeDisplay = () => {
    if (
      item.eligibilityInfo.structuredCriteria.minAgeInt === 0 &&
      item.eligibilityInfo.structuredCriteria.maxAgeInt > 120
    ) {
      return 'Not Specified';
    }
    if (
      item.eligibilityInfo.structuredCriteria.minAgeInt === 0 &&
      item.eligibilityInfo.structuredCriteria.maxAgeInt < 120
    ) {
      return `${item.eligibilityInfo.structuredCriteria.minAgeInt} years and younger`;
    }
    if (
      item.eligibilityInfo.structuredCriteria.minAgeInt > 0 &&
      item.eligibilityInfo.structuredCriteria.maxAgeInt < 120
    ) {
      return `${item.eligibilityInfo.structuredCriteria.minAgeInt} to ${item.eligibilityInfo.structuredCriteria.maxAgeInt} years`;
    }
    if (
      item.eligibilityInfo.structuredCriteria.minAgeInt > 0 &&
      item.eligibilityInfo.structuredCriteria.maxAgeInt > 120
    ) {
      return `${item.eligibilityInfo.structuredCriteria.minAgeInt} years and over`;
    }
  };

  const getLocationDisplay = () => {
    if (item.sites.length === 1) {
      const site = item.sites[0];
      let displayText = `${site.name}, ${site.city}, `;
      displayText +=
        site.country === 'United States'
          ? site.stateOrProvinceAbbreviation
          : site.country;
      return displayText;
    }
    // zip code present
    if (zip !== '') {
      //has a zip
      if (zipCoords.lat !== '' && zipCoords.lon !== '') {
        return `${item.sites.length} location${
          item.sites.length === 1 ? '' : 's'
        }, including ${countNearbySitesByZip(item.sites)} near you`;
      }
    }
    if (location === 'search-location-country') {
      return `${item.sites.length} location${
        item.sites.length === 1 ? '' : 's'
      }, including ${countNearbySitesByCountryParams(item.sites)} near you`;
    }
    return `${item.sites.length} location${item.sites.length === 1 ? '' : 's'}`;
  };

  return (
    <div className="results-list-item results-list__item">
      <div className="results-list-item__checkbox">
        <Checkbox
          id={id || item.nciID}
          name={item.nciID}
          checked={isChecked}
          label="Select this article for print"
          hideLabel
          onChange={() => onCheckChange(id)}
        />
      </div>
      <div className="results-list-item__contents">
        <div className="results-list-item__title">
          <Link
            to={`/about-cancer/treatment/clinical-trials/search/v?id=${item.nciID}`}
          >
            {item.briefTitle}
          </Link>
        </div>
        <div className="results-list-item__category">
          <span>Status:</span>
          {item.currentTrialStatus ? 'Active' : 'Active'}
        </div>
        <div className="results-list-item__category">
          <span>Age:</span>
          {getAgeDisplay()}
        </div>
        <div className="results-list-item__category">
          <span>Gender:</span>
          {getGenderDisplay(item.eligibilityInfo.structuredCriteria.gender)}
        </div>
        <div className="results-list-item__category">
          <span>Location:</span>
          {getLocationDisplay()}
        </div>
      </div>
    </div>
  );
};

ResultsListItem.propTypes = {
  id: PropTypes.string,
  item: PropTypes.object,
  isChecked: PropTypes.bool,
  onCheckChange: PropTypes.func.isRequired,
};

ResultsListItem.defaultProps = {
  results: [],
  isChecked: false,
};

export default ResultsListItem;
