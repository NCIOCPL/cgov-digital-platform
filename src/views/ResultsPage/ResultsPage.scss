.results-page {
  &__control {
    display: flex;
    justify-content: space-between;
    align-items: baseline;

    label {
      margin-top: 15px;
      margin-bottom: 0;
    }

    &.--top {
      margin-bottom: 20px;
      flex-direction: column-reverse;
      border-bottom: 1px solid #7d7d7d;
    }
    &.--bottom {
      flex-direction: column;
      border-top: 1px solid #7d7d7d;
    }

    @media screen and (min-width: 1025px) {
      flex-direction: row;
      &.--top {
        flex-direction: row;
      }
      &.--bottom {
        flex-direction: row;
        border-top: 1px solid #7d7d7d;
      }
    }
  }

  &__pager {
    border-bottom: 1px solid #7d7d7d;
    border-top: 1px solid #7d7d7d;
    padding: 5px 0;
    width: 100%;

    @media screen and (min-width: 1025px) {
      border: 0;
      width: auto;
    }
  }

  &__select-all {
    display: flex;
    align-items: baseline;
    padding: 16px 0;

    label:before {
      margin-right: 1.75rem;
    }
  }

  &__print-button {
    cursor: pointer;
    font-family: Montserrat, Avant Garde, Arial, sans-serif;
    font-weight: 700;
    line-height: normal;
    background-color: #006f8a;
    border: 0 #277c95;
    color: #fff;
    border-radius: 2px;
    margin: 0 28px 1.25em 0;
    padding: 5px 12px;
    font-size: 1em;
    min-height: 39px;
    margin: 0;
    &:hover {
      background: #00647c;
    }
  }

  &__content {
    @media screen and (min-width: 1025px) {
      display: flex;
      flex-direction: column;
      flex-wrap: nowrap;
      justify-content: space-between;
    }
  }

  &__list {
    display: flex;
  }

  &__aside {
    margin-bottom: 30px;
    display: none;
    @media screen and (min-width: 1025px) {
      flex: 0 0 24%;
      margin-left: 1%;
    }

    &.--side {
      @media screen and (min-width: 1025px) {
        display: flex;
        align-items: flex-start;
      }
    }
    &.--bottom {
      @media screen and (max-width: 1025px) {
        display: flex;
      }
    }
  }

  .cts-results-header {
    margin: 2em 0;

    h3 {
      margin-top: 0;
    }

    .reset-form {
      font-weight: bold;
    }
    .separator {
      margin: 0 5px;
    }
    .table-dropdown {
      margin: 0.5em 0 1em;
    }

    @media only screen and (min-width: 641px) {
      //tablet
      width: 90%;
    }

    @media screen and (min-width: 1025px) {
      //desktop
      width: 60%;
    }
  }

  // select all on page checkbox
  .check-all {
    margin-right: 1em;
    .cts-checkbox__label::before {
      margin-right: 0.75em;
    }
  }

  .cts-search-tip {
    display: flex;
    flex-wrap: nowrap;
    align-items: center;
    border: 1px solid #bdbdc2;
    padding: 16px;
    margin-bottom: 1em;

    &__icon {
      width: 3.9375em;
      height: 3.9375em;
      background-position: 7.8125% 52.90135%;
      min-width: 3.9375em;
      display: inline-block;
      background-size: 12em 262.5em;
    }
    &__body {
      padding-left: 1em;
    }
  }

  .cts-delighter-container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 100%;

    @media only screen and (min-width: 641px) {
      //tablet
      flex-direction: row;
      justify-content: space-around;
    }

    @media screen and (min-width: 1025px) {
      //desktop
      flex-direction: column;
      justify-content: center;
    }
  }
}

.btnAsLink {
  font-weight: 700;
}

.results-list {
  width: 100%;
  &__item {
    margin: 30px 0;
  }

  &.invalid-zip{
    li {
      position: relative;
      margin-top: 0.5em;
      margin-bottom: 0.5em;
      margin-left: 2em;
      &::before {
        content: '\2022';
        display: inline;
        position: absolute;
        left: -.75em;
        top: 0.25em;
        margin: 0;
        font-size: 1.1em;
        line-height: 0.90909em;
      }
    }
  }
}

.results-list-item {
  position: relative;
  display: flex;
  flex-direction: row;

  &__checkbox {
    position: relative;
    padding-top: 5px;
  }

  &__contents {
    width: 100%;
    margin: 0 1rem;
  }

  &__title {
    text-transform: capitalize;
  }

  &__category {
    span {
      font-weight: bold;
      margin-right: 5px;
    }
  }
}
