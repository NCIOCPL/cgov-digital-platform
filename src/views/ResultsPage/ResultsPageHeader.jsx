import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { useCachedValues } from '../../utilities/hooks';
import { getMainType, getCancerTypeDescendents } from '../../store/actions';
import { Link } from 'react-router-dom';
import { SearchCriteriaTable } from '../../components/atomic';
import { history } from '../../services/history.service';

const ResultsPageHeader = ({ handleUpdate, resultsCount }) => {
  const dispatch = useDispatch();
  const { formType, cancerType, age, zip, keywordPhrases } = useSelector(store => store.form);
  const {
    maintypeOptions = [],
  } = useCachedValues([
    'maintypeOptions',
  ]);

  const handleRefineSearch = () => {
    if (formType === 'basic') {
      //prefetch stuff
      if (maintypeOptions.length < 1) {
        dispatch(getMainType({}));
      }
      if(cancerType.name !== '') {
        handleUpdate('cancerTypeModified', true);
      }
      if (age !== ''){
        handleUpdate('ageModified', true);
      }
      if(zip !== ''){
        handleUpdate('zipModified', true);
      }
      if(keywordPhrases !== ''){
        handleUpdate('keywordPhrasesModified', true);
      }
      handleUpdate('formType', 'advanced');
    }
    handleUpdate('refineSearch', true);
    history.push('/search');
  };

  return (
    <div className="cts-results-header">
      <p>
        <strong>Results 1-10 of {resultsCount} for your search</strong>
      </p>

      <SearchCriteriaTable />
      <p className="reset-form">
        <Link to="/search">Start Over</Link>
        <span aria-hidden="true" className="separator">
          |
        </span>
        <button
          type="button"
          className="btnAsLink"
          onClick={handleRefineSearch}
        >
          Modify Search Criteria
        </button>
      </p>
    </div>
  );
};

ResultsPageHeader.propTypes = {
  resultsCount: PropTypes.string,
};

export default ResultsPageHeader;
