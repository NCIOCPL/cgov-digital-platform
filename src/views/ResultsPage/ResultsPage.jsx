import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import { Checkbox, ResultsList, Delighter, Pager } from '../../components/atomic';
import './ResultsPage.scss';

const ResultsPage = ({ results }) => {
  const [selectAll, setSelectAll] = useState(false);
  const [paginatedResults, setPaginatedResults] = useState([]);
  const [selectedResults, setSelectedResults] = useState([]);

  useEffect(() => {
    if (selectAll) {
      setSelectedResults([...paginatedResults.map(result => result.title)]);
    } else {
      setSelectedResults([]);
    }
  }, [selectAll]);

  useEffect(() => {
    setSelectedResults([]);
  }, [paginatedResults]);

  useEffect(() => {
    if (selectedResults.length === 0) {
      setSelectAll(false);
    } else if (paginatedResults.length === selectedResults.length) {
      setSelectAll(true);
    }
  }, [paginatedResults, selectedResults]);

  const handlePagination = (slicedResults) => {
    setPaginatedResults([...slicedResults]);
  }

  const renderDelighters = () => (
    <div className="cts-delighter-container">
      <Delighter
        classes="cts-livehelp"
        url="/contact"
        titleText={
          <>
            Have a question?
            <br />
            We're here to help
          </>
        }
      >
        <p>
          <strong>Chat with us:</strong> LiveHelp
          <br />
          <strong>Call us:</strong> 1-800-4-CANCER
          <br />
          (1-800-422-6237)
        </p>
      </Delighter>

      <Delighter
        classes="cts-what"
        url="/about-cancer/treatment/clinical-trials/what-are-trials"
        titleText={<>What Are Cancer Clinical Trials?</>}
      >
        <p>Learn what they are and what you should know about them.</p>
      </Delighter>

      <Delighter
        classes="cts-which"
        url="/about-cancer/treatment/clinical-trials/search/trial-guide"
        titleText={<>Which trials are right for you?</>}
      >
        <p>
          Use the checklist in our guide to gather the information youâ€™ll need.
        </p>
      </Delighter>
    </div>
  );

  return (
    <div className="general-page-body-container">
      <div className="contentzone">
        {/* */}
        <article className="results-page">
          <div className="results-page__control">
            <Checkbox
              id="select-all-checkbox"
              name="select-all"
              label="Select All on Page"
              classes="results-page__select-all"
              checked={selectAll}
              onChange={() => setSelectAll(!selectAll)}
            />
            <Pager data={results} callback={handlePagination}/>
          </div>
          <div className="results-page__content">
            {
              <ResultsList
                results={paginatedResults}
                selectedResults={selectedResults}
                setSelectedResults={setSelectedResults}
              />
            }
            <aside className="results-page__aside">{renderDelighters()}</aside>
          </div>
        </article>
        {/* */}
      </div>
    </div>
  );
};

ResultsPage.propTypes = {
  results: PropTypes.array,
};

ResultsPage.defaultProps = {
  results: [],
};

export default ResultsPage;
