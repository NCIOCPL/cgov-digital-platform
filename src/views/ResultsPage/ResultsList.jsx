import React from 'react';
import PropTypes from 'prop-types';
import ResultsListItem from './ResultsListItem';


const ResultsList = ({ results, selectedResults, setSelectedResults, setSelectAll }) => {
  const handleOnCheckChange = id => {
    if (selectedResults.indexOf(id) === -1) {
      setSelectedResults([...selectedResults, id]);
    } else {
      // remove from selected
      setSelectAll(false);
      setSelectedResults(selectedResults.filter(item => item !== id));
    }
  };

  return (
    <div className="results-list">
      {results.map(item => {
        return (
          <ResultsListItem
            key={item.nciID}
            id={item.nciID}
            item={item}
            isChecked={selectedResults.indexOf(item.nciID) > -1}
            onCheckChange={handleOnCheckChange}
          />
        );
      })}
    </div>
  );
};

ResultsList.propTypes = {
  setSelectedResults: PropTypes.func,
  selectedResults: PropTypes.array,
  results: PropTypes.array
};

ResultsList.defaultProps = {
  results: [],
  selectedResults: []
};

export default ResultsList;
