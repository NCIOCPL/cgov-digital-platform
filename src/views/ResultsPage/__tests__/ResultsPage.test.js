import React from 'react';
import { render } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import ResultsPage from '../ResultsPage';

const defaultProps = {};

const setup = (enzymeMethod = render, props = {}) => {
  const component = enzymeMethod(
    <MemoryRouter>
      <ResultsPage {...props} />
    </MemoryRouter>
  );
  return component;
};

describe('ResultsPage', () => {
  describe('Render', () => {
    it('renders without error', () => {
      const component = setup(render, defaultProps);
      expect(component).toMatchSnapshot();
    });
  });
});
