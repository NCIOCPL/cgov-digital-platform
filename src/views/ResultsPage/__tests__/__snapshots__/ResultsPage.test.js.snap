// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsPage Render renders without error 1`] = `
<div
  class="general-page-body-container"
>
  <div
    class="contentzone"
  >
    <article
      class="results-page"
    >
      <div
        class="cts-results-header"
      >
        <p>
          <strong>
            Results 1-10 of 0 for your search
          </strong>
        </p>
        <div
          class="cts-accordion cts-accordion--bordered"
        >
          <h2
            aria-expanded="true"
            class="cts-accordion__heading"
          >
            <button
              aria-controls="-content"
              aria-expanded="true"
              class="cts-accordion__button"
            >
              <span>
                Show Search Criteria
              </span>
            </button>
          </h2>
          <div
            aria-hidden="false"
            class="cts-accordion__content"
            id="-content"
          >
            <div>
              <h3>
                Your Search Criteria
              </h3>
              ...Table here...
            </div>
          </div>
        </div>
        <p
          class="reset-form"
        >
          <a
            href="/search"
          >
            Start Over
          </a>
        </p>
      </div>
      <div
        class="results-page__content"
      >
        <div
          class="results-page__control --top"
        >
          <div
            class="results-page__select-all"
          >
            <div
              class="cts-checkbox "
            >
              <input
                class="cts-checkbox__input"
                id="select-all-checkbox"
                name="select-all"
                type="checkbox"
                value="select-all-checkbox"
              />
              <label
                class="cts-checkbox__label"
                for="select-all-checkbox"
              >
                Select All on Page
              </label>
            </div>
            <button
              class="results-page__print-button"
            >
              Print Selected
            </button>
          </div>
          <div
            class="results-page__pager"
          >
            <nav
              class="pager__container"
            />
          </div>
        </div>
        <div
          class="results-page__list"
        >
          <div
            class="results-list"
          />
          <aside
            class="results-page__aside --side"
          >
            <div
              class="cts-delighter-container"
            >
              <div
                class="delighter cts-livehelp"
              >
                <a
                  href="/contact"
                >
                  <h4>
                    Have a question?
                    <br />
                    We're here to help
                  </h4>
                  <p>
                    <strong>
                      Chat with us:
                    </strong>
                     LiveHelp
                    <br />
                    <strong>
                      Call us:
                    </strong>
                     1-800-4-CANCER
                    <br />
                    (1-800-422-6237)
                  </p>
                </a>
              </div>
              <div
                class="delighter cts-which"
              >
                <a
                  href="/about-cancer/treatment/clinical-trials/search/trial-guide"
                >
                  <h4>
                    Which trials are right for you?
                  </h4>
                  <p>
                    Use the checklist in our guide to gather the information you’ll need.
                  </p>
                </a>
              </div>
            </div>
          </aside>
        </div>
        <div
          class="results-page__control --bottom"
        >
          <div
            class="results-page__select-all"
          >
            <div
              class="cts-checkbox "
            >
              <input
                class="cts-checkbox__input"
                id="select-all-checkbox"
                name="select-all"
                type="checkbox"
                value="select-all-checkbox"
              />
              <label
                class="cts-checkbox__label"
                for="select-all-checkbox"
              >
                Select All on Page
              </label>
            </div>
            <button
              class="results-page__print-button"
            >
              Print Selected
            </button>
          </div>
          <div
            class="results-page__pager"
          >
            <nav
              class="pager__container"
            />
          </div>
        </div>
      </div>
      <aside
        class="results-page__aside --bottom"
      >
        <div
          class="cts-delighter-container"
        >
          <div
            class="delighter cts-livehelp"
          >
            <a
              href="/contact"
            >
              <h4>
                Have a question?
                <br />
                We're here to help
              </h4>
              <p>
                <strong>
                  Chat with us:
                </strong>
                 LiveHelp
                <br />
                <strong>
                  Call us:
                </strong>
                 1-800-4-CANCER
                <br />
                (1-800-422-6237)
              </p>
            </a>
          </div>
          <div
            class="delighter cts-which"
          >
            <a
              href="/about-cancer/treatment/clinical-trials/search/trial-guide"
            >
              <h4>
                Which trials are right for you?
              </h4>
              <p>
                Use the checklist in our guide to gather the information you’ll need.
              </p>
            </a>
          </div>
        </div>
      </aside>
    </article>
  </div>
</div>
`;
